src/mongoose_scram.erl

```diff
@@ -32,20 +32,34 @@
 
 -export([serialize/1, deserialize/1]).
 
--export([scram_to_tuple/1]).
+-export([scram_to_tuple/1, scram_record_to_map/1]).
 
 -type sha_type() :: crypto:sha1() | crypto:sha2().
 
 -type scram_tuple() :: { StoredKey :: binary(), ServerKey :: binary(),
                          Salt :: binary(), Iterations :: non_neg_integer() }.
 
+-type scram_map() ::
+    #{salt := binary(),
+      iteration_count := non_neg_integer(),
+      sha := #{server_key := binary(), stored_key := binary()}}.
+
+-type muilti_scram_map() ::
+    #{salt := binary(),
+      iteration_count := non_neg_integer(),
+      sha    := #{server_key := binary(), stored_key := binary()},
+      sha256 := #{server_key := binary(), stored_key := binary()}}.
+
```

Comment:
I think we can have only one type for that with `sha*` keys marked as optional
```suggestion
-type scram_map() ::
    #{salt := binary(),
      iteration_count := non_neg_integer(),
      sha    => #{server_key := binary(), stored_key := binary()},
      sha256 => #{server_key := binary(), stored_key := binary()}}.
```
What do you think?